x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_initializer.tsx

```diff
@@ -0,0 +1,155 @@
+import { MlJob } from '@elastic/elasticsearch/lib/api/types';
+import type { SingleMetricViewerServices } from '..';
+import { TimeRangeBounds } from '../../application/util/time_buckets';
+import { SeriesControls } from '../../application/timeseriesexplorer/components/series_controls';
+import {
+  APP_STATE_ACTION,
+  type TimeseriesexplorerActionType,
+} from '../../application/timeseriesexplorer/timeseriesexplorer_constants';
+
+export interface SingleMetricViewerInitializerProps {
+  bounds: TimeRangeBounds;
+  defaultTitle: string;
+  initialInput?: SingleMetricViewerServices;
+  job: MlJob;
+  onCreate: (props: {
+    panelTitle: string;
+    functionDescription: string;
+    selectedDetectorIndex: number;
+    selectedEntities: any;
+  }) => void;
+  onCancel: () => void;
+}
+
+export const SingleMetricViewerInitializer: FC<SingleMetricViewerInitializerProps> = ({
+  bounds,
+  defaultTitle,
+  initialInput,
+  job,
+  onCreate,
+  onCancel,
+}) => {
+  const [panelTitle, setPanelTitle] = useState<string>(defaultTitle);
+  const [functionDescription, setFunctionDescription] = useState<string>('');
+  const [selectedDetectorIndex, setSelectedDetectorIndex] = useState<number>(0);
+  const [selectedEntities, setSelectedEntities] = useState<any>();
+
+  const isPanelTitleValid = panelTitle.length > 0;
+
+  const handleStateUpdate = (action: TimeseriesexplorerActionType, payload: any) => {
+    switch (action) {
+      case APP_STATE_ACTION.SET_ENTITIES:
+        setSelectedEntities(payload);
+        break;
+      case APP_STATE_ACTION.SET_FUNCTION_DESCRIPTION:
+        setFunctionDescription(payload);
+        break;
+      case APP_STATE_ACTION.SET_DETECTOR_INDEX:
+        setSelectedDetectorIndex(payload);
+        break;
+      default:
+        break;
+    }
+  };
+
+  return (
+    <EuiModal
+      initialFocus="[name=panelTitle]"
+      onClose={onCancel}
+      data-test-subj={'mlSingleMetricViewerEmbeddableInitializer'}
+    >
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          <FormattedMessage
+            id="xpack.ml.SingleMetricViewerEmbeddable.setupModal.title"
+            defaultMessage="Single metric viewer configuration"
+          />
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+
+      <EuiModalBody>
+        <EuiForm>
+          <EuiFormRow
+            label={
+              <FormattedMessage
+                id="xpack.ml.singleMetricViewerEmbeddable.panelTitleLabel"
+                defaultMessage="Panel title"
+              />
+            }
+            isInvalid={!isPanelTitleValid}
+          >
+            <EuiFieldText
+              data-test-subj="panelTitleInput"
+              id="panelTitle"
+              name="panelTitle"
+              value={panelTitle}
+              onChange={(e) => setPanelTitle(e.target.value)}
+              isInvalid={!isPanelTitleValid}
+            />
+          </EuiFormRow>
+
+          <SeriesControls
```

Comment:
What is the 'Function' dropdown for in the modal? Doesn't the detector config have all we need to plot the correct chart?

<img width="605" alt="image" src="https://github.com/elastic/kibana/assets/7405507/89f8de5b-031e-4d44-bd17-9902f5608689">

Reply:
Fixed in d931543fa904ca44d10ca6e19a0855dab00062d6

Reply:
This is fixed for all detectors, except where the `metric` function is used in the detector (can only be created in the advanced wizard or with the API). For detectors using `metric` the following is displayed when the modal is open and a split field is selected:

<img width="305" alt="image" src="https://github.com/elastic/kibana/assets/7405507/cedb1c75-75ea-4c85-9d06-63271ced1735">

and on applying the config the 'No results found' message is displayed in the panel.

Seeing as there are unrelated issues with the `metric` detector in the Single Metric Viewer - github.com/elastic/kibana/issues/176301, the simplest approach may be to add `metric` to the list of functions which aren't visible in the SMV and address this embeddable as part of a fix for that issue.

Reply:
@peteharverson 

Added the fix for the SMV issue with metric jobs for both the ML plugin view and in the dashboard in 4b514c3195216a59eb8f058b8bf1b4c5cbb1b0db

Both should be viewable now 👍